[icu] build failure icu 70.1, error: Host path separator (:) in path; this is unsupported.

**Host Environment**
- OS: [centos 7.9]
- Compiler: revision

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install icu:x64-linux`

**Failure logs**
[root@centos-dev vcpkg]# ./vcpkg install icu:x64-linux
Computing installation plan...
The following packages will be built and installed:
    icu[core]:x64-linux -> 70.1
Detecting compiler hash for triplet x64-linux...
Restored 0 packages from /root/.cache/vcpkg/archives in 5.45 us. Use --debug to see more details.
Starting package 1/1: icu:x64-linux
Building package icu[core]:x64-linux...
-- Using cached icu4c-70_1-src.tgz.
-- Cleaning sources at /usr/work/vcpkg/buildtrees/icu/src/c-70_1-src-a6ee276eef.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /usr/work/vcpkg/downloads/icu4c-70_1-src.tgz
-- Applying patch disable-escapestr-tool.patch
-- Applying patch remove-MD-from-configure.patch
-- Applying patch fix_parallel_build_on_windows.patch
-- Applying patch fix-extra.patch
-- Applying patch mingw-dll-install.patch
-- Applying patch disable-static-prefix.patch
-- Applying patch fix-win-build.patch
-- Using source at /usr/work/vcpkg/buildtrees/icu/src/c-70_1-src-a6ee276eef.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
CMake Error at scripts/cmake/vcpkg_host_path_list.cmake:43 (message):
  Host path separator (:) in path; this is unsupported.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_host_path_list.cmake:72 (z_vcpkg_translate_to_host_path_list)
  scripts/cmake/vcpkg_configure_make.cmake:218 (vcpkg_host_path_list)
  scripts/cmake/vcpkg_configure_make.cmake:548 (z_prepend_include_path)
  ports/icu/portfile.cmake:52 (vcpkg_configure_make)
  scripts/ports.cmake:145 (include)


error: building icu:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+icu
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[icu]+Build+error
including:
  package: icu[core]:x64-linux -> 70.1
    vcpkg-tool version: 2022-03-30-692785ac944e81417840c6de244fb3e18a4b35eb
    vcpkg-scripts version: 99346bb69 2022-04-13 (14 hours ago)

Additionally, attach any relevant sections from the log files above.

***@***.*** vcpkg]# ./vcpkg install icu:x64-linux
Computing installation plan...
The following packages will be built and installed:
&nbsp; &nbsp; icu[core]:x64-linux -&gt; 71.1
Detecting compiler hash for triplet x64-linux...
Restored 0 packages from /root/.cache/vcpkg/archives in 16.67 us. Use --debug to see more details.
Starting package 1/1: icu:x64-linux
Building package icu[core]:x64-linux...
-- Using cached icu4c-71_1-src.tgz.
-- Cleaning sources at /usr/work/vcpkg/buildtrees/icu/src/c-71_1-src-08c83f7de1.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /usr/work/vcpkg/downloads/icu4c-71_1-src.tgz
-- Applying patch disable-escapestr-tool.patch
-- Applying patch remove-MD-from-configure.patch
-- Applying patch fix_parallel_build_on_windows.patch
-- Applying patch fix-extra.patch
-- Applying patch mingw-dll-install.patch
-- Applying patch disable-static-prefix.patch
-- Applying patch fix-win-build.patch
-- Applying patch check-autoconf-archive.patch
-- Using source at /usr/work/vcpkg/buildtrees/icu/src/c-71_1-src-08c83f7de1.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
CMake Error at scripts/cmake/vcpkg_host_path_list.cmake:43 (message):
&nbsp; Host path separator (:) in path; this is unsupported.
Call Stack (most recent call first):
&nbsp; scripts/cmake/vcpkg_host_path_list.cmake:72 (z_vcpkg_translate_to_host_path_list)
&nbsp; scripts/cmake/vcpkg_configure_make.cmake:218 (vcpkg_host_path_list)
&nbsp; scripts/cmake/vcpkg_configure_make.cmake:549 (z_prepend_include_path)
&nbsp; ports/icu/portfile.cmake:53 (vcpkg_configure_make)
&nbsp; scripts/ports.cmake:146 (include)




error: building icu:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
&nbsp; https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+icu
You can submit a new issue at:
&nbsp; https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&amp;title=[icu]+Build+error
including:
&nbsp; package: icu[core]:x64-linux -&gt; 71.1
&nbsp; &nbsp; vcpkg-tool version: 2022-03-30-692785ac944e81417840c6de244fb3e18a4b35eb
&nbsp; &nbsp; vcpkg-scripts version: 0d7603c4e 2022-05-09 (66 minutes ago)


Additionally, attach any relevant sections from the log files above.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/vcpkg"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月10日(星期二) 下午2:48
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/vcpkg] [icu] build failure icu 70.1, error: Host path separator (:) in path; this is unsupported. (Issue #24138)





 
@chris-shenzhen-china Failure logs please?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***&gt;

--- a/docs/maintainers/vcpkg_configure_make.md
+++ b/docs/maintainers/vcpkg_configure_make.md
@@ -16,6 +16,7 @@ vcpkg_configure_make(
     [CONFIG_DEPENDENT_ENVIRONMENT <SOME_VAR>...]
     [CONFIGURE_ENVIRONMENT_VARIABLES <SOME_ENVVAR>...]
     [ADD_BIN_TO_PATH]
+    [DISABLE_VERBOSE_FLAGS]
     [NO_DEBUG]
     [SKIP_CONFIGURE]
     [PROJECT_SUBPATH <${PROJ_SUBPATH}>]
@@ -59,8 +60,8 @@ Script that needs to be called before configuration (do not use for batch files 
 ### ADD_BIN_TO_PATH
 Adds the appropriate Release and Debug `bin\` directories to the path during configure such that executables can run against the in-tree DLLs.
 
-## DISABLE_VERBOSE_FLAGS
-do not pass '--disable-silent-rules --verbose' to configure
+### DISABLE_VERBOSE_FLAGS
+Do not pass '--disable-silent-rules --verbose' to configure.
 
 ### OPTIONS
 Additional options passed to configure during the configuration.

--- a/scripts/cmake/vcpkg_configure_make.cmake
+++ b/scripts/cmake/vcpkg_configure_make.cmake
@@ -15,6 +15,7 @@ vcpkg_configure_make(
     [CONFIG_DEPENDENT_ENVIRONMENT <SOME_VAR>...]
     [CONFIGURE_ENVIRONMENT_VARIABLES <SOME_ENVVAR>...]
     [ADD_BIN_TO_PATH]
+    [DISABLE_VERBOSE_FLAGS]
     [NO_DEBUG]
     [SKIP_CONFIGURE]
     [PROJECT_SUBPATH <${PROJ_SUBPATH}>]
@@ -58,8 +59,8 @@ Script that needs to be called before configuration (do not use for batch files 
 ### ADD_BIN_TO_PATH
 Adds the appropriate Release and Debug `bin\` directories to the path during configure such that executables can run against the in-tree DLLs.
 
-## DISABLE_VERBOSE_FLAGS
-do not pass '--disable-silent-rules --verbose' to configure
+### DISABLE_VERBOSE_FLAGS
+Do not pass '--disable-silent-rules --verbose' to configure.
 
 ### OPTIONS
 Additional options passed to configure during the configuration.

--- a/scripts/cmake/vcpkg_configure_make.cmake
+++ b/scripts/cmake/vcpkg_configure_make.cmake
@@ -208,19 +208,6 @@ macro(z_vcpkg_append_to_configure_environment inoutstring var defaultval)
     endif()
 endmacro()
 
-# Setup include environment (since these are buildtype independent restoring them is unnecessary)
-macro(z_prepend_include_path var)
-    unset(ENV{${var}})
-    if(NOT DEFINED z_vcpkg_env_backup_${var} OR "${z_vcpkg_env_backup_${var}}" STREQUAL "")
-        vcpkg_host_path_list(APPEND ENV{${var}} "${CURRENT_INSTALLED_DIR}/include")
-    else()
-        foreach (one_bk IN ITEMS ${z_vcpkg_env_backup_${var}})
-            vcpkg_host_path_list(PREPEND ENV{${var}} "${one_bk}")
-        endforeach()
-        vcpkg_host_path_list(PREPEND ENV{${var}} "${CURRENT_INSTALLED_DIR}/include")
-    endif()
-endmacro()
-
 macro(z_convert_to_list input output)
     string(REGEX MATCHALL "(( +|^ *)[^ ]+)" ${output} "${${input}}")
 endmacro()
@@ -544,10 +531,10 @@ function(vcpkg_configure_make)
     endif()
 
     # Used by CL 
-    z_prepend_include_path(INCLUDE)
+    vcpkg_host_path_list(PREPEND ENV{INCLUDE} "${CURRENT_INSTALLED_DIR}/include")
     # Used by GCC
-    z_prepend_include_path(C_INCLUDE_PATH)
-    z_prepend_include_path(CPLUS_INCLUDE_PATH)
+    vcpkg_host_path_list(PREPEND ENV{C_INCLUDE_PATH} "${CURRENT_INSTALLED_DIR}/include")
+    vcpkg_host_path_list(PREPEND ENV{CPLUS_INCLUDE_PATH} "${CURRENT_INSTALLED_DIR}/include")
 
     # Flags should be set in the toolchain instead (Setting this up correctly requires a function named vcpkg_determined_cmake_compiler_flags which can also be used to setup CC and CXX etc.)
     if(VCPKG_TARGET_IS_WINDOWS)

